### ISSUE TITLE
imp7,8

### ISSUE THREAD

**Issue by THESAURABH07** [THESAURABH07/data-main#5]
7. import numpy as np
import matplotlib.pyplot as plt
from sklearn.datasets import load_iris
from sklearn.neighbors import
KNeighborsClassifier
from sklearn.model_selection import
train_test_split
# Load Iris dataset and use only 2 features
iris = load_iris()
X = iris.data[:, :2] # Sepal length & width
y = iris.target
# Split data
X_train, X_test, y_train, y_test = train_test_split(X,
y, test_size=0.2, random_state=42)
# Fit KNN
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train, y_train)
# Create mesh grid
x_min, x_max = X[:, 0].min() - 1, X[:, 0].max() + 1
y_min, y_max = X[:, 1].min() - 1, X[:, 1].max() + 1
xx, yy = np.meshgrid(np.arange(x_min, x_max,
0.02),np.arange(y_min, y_max, 0.02))
# Predict over grid
Z = knn.predict(np.c_[xx.ravel(), yy.ravel()])
Z = Z.reshape(xx.shape)
# Plot
plt.figure(figsize=(8, 6))
plt.contourf(xx, yy, Z, cmap=plt.cm.Pastel2,
alpha=0.8)
plt.scatter(X_train[:, 0], X_train[:, 1], c=y_train,
edgecolors='k', label='Train', cmap=plt.cm.Set1)
plt.scatter(X_test[:, 0], X_test[:, 1], c=y_test,
marker='x', label='Test', cmap=plt.cm.Set1)
plt.xlabel("Sepal length")
plt.ylabel("Sepal width")
plt.title("KNN Decision Boundary (k=5)")
plt.legend()
plt.grid(True)
plt.show()
====================================
======================
#program 8-Write a program to implement a linear
regression model for regression tasks and Train
#the model on a dataset with continuous target
variables
import numpy as np
import matplotlib.pyplot as plt
from sklearn.datasets import
fetch_california_housing
from sklearn.linear_model import
LinearRegression
# Load data
data = fetch_california_housing()
X = data.data[:, [0]] # Use only 'MedInc' feature
(column 0)
y = data.target
# Train linear regression model
model = LinearRegression()
model.fit(X, y)
# Predict line
x_range = np.linspace(X.min(), X.max(),
100).reshape(-1, 1)
y_pred_line = model.predict(x_range)
# Plot data and regression line
plt.figure(figsize=(8, 5))
plt.scatter(X, y, alpha=0.3, label="Data",
color='skyblue')
plt.plot(x_range, y_pred_line, color='red',
label="Best Fit Line")
plt.title("Linear Regression: Median Income vs
House Value")
plt.xlabel("Median Income")
plt.ylabel("House Value (in $100,000s)")
plt.legend()
plt.grid(True)
plt.show()
